lua/telescope/config.lua

```diff
@@ -111,19 +111,25 @@ local layout_config_description = string.format([[
 
 local telescope_defaults = {
 
-  sorting_strategy = {
-    "descending",
-    [[
+  --TODO(conni2461): This is kinda aweful. Why because we cant have a deterministic
+  --                 order. So we have to sort it which makes
+  --                 `buffer_previewer_maker` the first option, it should be the
+  --                 last because its a developer option. This needs to be a
+  --                 list with a set order so we dont have to sort it.
+  --
+  --                 Or we go back to something similar to what we had before.
+  --                 We do it with a function that will make two tables, one
+  --                 that results in this table, and a second one that keeps
+  --                 track of the order. We do something similar in docgen.
```

Comment:
CC @l-kershaw thoughts ???

Reply:
Reading over my comments. Its not good :laughing: 
I am refering to the generated documentation: https://github.com/nvim-telescope/telescope.nvim/blob/01366e2ef7f320730ba9fcdace92835fb53376dc/doc/telescope.txt#L25

I like to have full control over the order in which the keys are documented. With the current way of just having a table with `key = { value, docs }` we do not have a deterministic order if we iterate over the table (thats because how lua works). So we collect the keys and sort the keys. That might lead to more unimportant keys (for new users) pop up as one of the first keys. Does this make sense to you?

Reply:
I agree, it would be better for us to be able to specify a set order for our defaults.
That way we can group related things together, and put the more commonly used options nearer the top.

It shouldn't be too difficult to implement either, it _looks_ like it just requires a change of the use of `set` in `config.set_defaults` and some tweaking in `telescope.__format_setup_keys`.
(_that being said, all of the documentation generation is wizardry to me at the moment, so I could be wrong 🤣_)

Reply:
I was just asking you because you introduced it :laughing: I can do it its not that hard, we just need to tweak everything a little bit :laughing: 

`docgen` is actually lua magic :laughing: with a lot of treesitter ;)